### PR TITLE
[feat](Nereids): eliminate inner join by pk fk when comparing mv

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
add some regression case, please

**Comment by XieJiann** [comment]
> add some regression case, please

this has not been finished, I'll add some test in it

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java [hunk 1]
```diff
@@ -279,11 +282,46 @@ private static List<Slot> applyNullable(List<Slot> slots, boolean nullable) {
...
+        }
+
+        ForeignKeyContext context = new ForeignKeyContext();
+        context.collectForeignKeyConstraint(primaryPlan);
+        context.collectForeignKeyConstraint(foreignPlan);
+
+        ImmutableEqualSet<Slot> equalSet = join.getEqualSlots();
+        Set<Slot> primaryKey = Sets.intersection(equalSet.getAllItemSet(), primaryPlan.getOutputSet());
+        Set<Slot> foreignKey = Sets.intersection(equalSet.getAllItemSet(), foreignPlan.getOutputSet());
+        if (!context.isForeignKey(foreignKey) && !context.isPrimaryKey(primaryKey)) {
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java:313]: maybe this should be ||（or）?

**Review by seawinde — APPROVED** [review]

**Comment by XieJiann** [comment]
run buildall

**Review by jackwener — APPROVED** [review]

**Pull request merged**